New page for pub bump (#6196)
dart-lang/site-www: src/content/tools/pub/cmd/pub-bump.md

---
title: dart pub bump
description: Use dart pub bump to increase the version number of the current package.
---

_Bump_ is one of the commands of the [pub tool](/tools/pub/cmd).

```plaintext

> $ dart pub bump <subcommand> [arguments]

Reviewer: The command currently doesn't have any arguments you can pass
```suggestion
$ dart pub bump <subcommand>
```

```

> This command increments the verion number of the current package.

Reviewer: ```suggestion
This command increments the version number of the current package.
```

> It allows the subcommands `breaking`, `major`, `minor`,
> and `patch` to increment only a specific part of the version number. 

Reviewer: ```suggestion
Use one of the subcommands `breaking`, `major`, `minor`,
or `patch` according to the type of increment desired. 

Refer to [semver](https://semver.org/spec/v2.0.0-rc.1.html) for guidance on versioning your package.
```

For example:

```console
$ dart pub bump minor
  Updating version from 1.0.0 to 1.1.0
  
  Diff:
  - version: 1.0.0
  + version: 1.1.0
  
  Remember to update `CHANGELOG.md` before publishing.
```

## Options

For options that apply to all pub commands, see
[Global options](/tools/pub/cmd#global-options).

### `-n, --dry-run`

Reports what would change for the version number, and shows the version diff,

> without changing anything.

Reviewer: This is not actually implemented (perhaps it should be though...)

Author: Oh `dry-run` isn't implemented? [It comes up a few times in the PR though](https://github.com/dart-lang/pub/pull/4361/files#diff-f03751343a35e73516dca2f40772206134537c925f66d0505cc6c5c28e0b9976R25), is that just like making space for it in the future?

I'll remove it anyway for now to land the page, just curious!

Reviewer: Oh I'm sorry - you are totally right.

I was looking at `dart pub bump --help` but should of course look at `dart pub bump <subcommand> help`.

And it is indeed implemented (by me...).

Thanks for pointing this out!

Author: No worries! Thanks for explaining, I put it back and specified it's for each subcommand
